File: TPAppMobiles/imagebutton/src/main/java/com/example/imagebutton/MainActivity.java
package com.example.imagebutton;

import androidx.appcompat.app.AppCompatActivity;

import android.media.Image;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageButton;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    TextView textView;
    ImageButton imageButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        imageButton = findViewById(R.id.imageButton);
        imageButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                textView = findViewById(R.id.textView);
                textView.setText(R.string.nouvTexte);
            }
        });
    }
}

File: KotlinApps/imageBouton/src/main/java/com/example/kotlinapps/MainActivity.kt
package com.example.kotlinapps

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.ImageButton
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var texteBouton = findViewById<TextView>(R.id.texteBouton)
        var imageBouton = findViewById<ImageButton>(R.id.imageBouton)

        imageBouton.setOnClickListener(View.OnClickListener { texteBouton.setText(imageBouton.contentDescription) })
    }

}